pkg-new/preflights/run.go

```diff
@@ -15,30 +15,53 @@ import (
 	apitypes "github.com/replicatedhq/embedded-cluster/api/types"
 	ecv1beta1 "github.com/replicatedhq/embedded-cluster/kinds/apis/v1beta1"
 	"github.com/replicatedhq/embedded-cluster/pkg/helpers"
-	"github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig"
 	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
 	"sigs.k8s.io/yaml"
 )
 
-// Run runs the provided host preflight spec locally. This function is meant to be
-// used when upgrading a local node.
-func (p *PreflightsRunner) Run(ctx context.Context, spec *troubleshootv1beta2.HostPreflightSpec, rc runtimeconfig.RuntimeConfig) (*apitypes.HostPreflightsOutput, string, error) {
+// RunHostPreflights runs the provided host preflight spec locally.
+func (p *PreflightsRunner) RunHostPreflights(ctx context.Context, spec *troubleshootv1beta2.HostPreflightSpec, opts RunOptions) (*apitypes.PreflightsOutput, string, error) {
```

Comment:
I think the methods in this file are largely untested right?

We should consider setting up some `os/exec` `Cmd` interface - https://pkg.go.dev/os/exec#Cmd - we can mock in tests and test the remaining logic around the `run` methods.

Reply:
agreed. ok if it's a separate pr though? feels like something this package is lacking prior to this pr.